pkg/oracle_node.go

```diff
@@ -95,30 +96,19 @@ func NewOracleNode(ctx context.Context, privKey crypto.PrivKey, portNbr int, use
 		return nil, err
 	}
 
-	// Extract the public key from the private key
-	pubKey := privKey.GetPublic()
-	if err != nil {
-		return nil, err
-	}
-
-	// Pass the public key as the third argument to NewPubSubManager
-	subscriptionManager, err := pubsub2.NewPubSubManager(ctx, hst, pubKey)
+	subscriptionManager, err := pubsub2.NewPubSubManager(ctx, hst)
 	if err != nil {
 		return nil, err
 	}
 
-	ecdsaPrivKey, err := crypto2.Libp2pPrivateKeyToEcdsa(privKey)
-	if err != nil {
-		return nil, err
-	}
 	return &OracleNode{
 		Host:          hst,
-		PrivKey:       ecdsaPrivKey,
-		Protocol:      ProtocolWithVersion(oracleProtocol),
+		PrivKey:       crypto2.KeyManagerInstance().EcdsaPrivKey,
```

Comment:
Remind me @restevens402 why we use the EcdsaPrivKey here? I got a blank.

Reply:
It looks like it is not used.  We added it for a reason. I am guessing it was related to how we were dealing with the public key. It looks like this can be removed as an attribute of the node.